src/main/java/com/transportcompany/transportcompany/controllers/UserController.java

```diff
@@ -1,36 +1,46 @@
 package com.transportcompany.transportcompany.controllers;
+
 import com.transportcompany.transportcompany.models.dtos.UserDTO;
 import com.transportcompany.transportcompany.models.entities.User;
 import com.transportcompany.transportcompany.repositories.UserRepository;
+import com.transportcompany.transportcompany.services.UserService;
+import jakarta.validation.Valid;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+
 import java.util.List;
 import java.util.Optional;
 
 
 @RestController
 public class UserController {
 
-    @GetMapping ("/api/users")
-    public List <UserDTO>getUsers() {return userService.getUsers();
+    @Autowired
+    private final UserService userService;
+    public UserController(UserService userService) {
+        this.userService = userService;
     }
 
-    @Autowired
-    private UserRepository userRepository;
-    @GetMapping("")
-    public List<User> getAllUsers() {
-        return userRepository.findAll();
+    @PostMapping("/api/users")
+    public  ResponseEntity<UserDTO> createUser(@RequestBody UserDTO userDTO) {
+        return ResponseEntity.ok(userService.createUser(userDTO));
+    }
+
+
+
+    @GetMapping("/api/users")
+    public List<UserDTO> getUsers() {
```

Comment:
If you are using `ResponseEntity` you should use it on all controller methods.

Reply:
Solve